xtask/.gitignore

```diff
@@ -0,0 +1,2 @@
+Cargo.lock
```

Comment:
thought: We _may_ want to keep the `Cargo.lock`, but I feel like it's annoying to do so, with no real gain here.

Reply:
Actually, this may be beneficial specifically for [caching with `rust-cache`](https://github.com/Swatinem/rust-cache#cache-effectiveness):

> It is also most effective for repositories with a `Cargo.lock` file. Library repositories with only a `Cargo.toml` file have limited benefits, as cargo will _always_ use the most up-to-date dependency versions, which may not be cached.